src/cpp/fastdds/subscriber/DataReaderImpl.cpp

```diff
@@ -913,6 +913,19 @@ void DataReaderImpl::InnerDataReaderListener::onReaderMatched(
         const SubscriptionMatchedStatus& info)
 {
     data_reader_->update_subscription_matched_status(info);
+
+    StatusMask notify_status = StatusMask::subscription_matched();
+    DataReaderListener* listener = data_reader_->get_listener_for(notify_status);
+    if (listener != nullptr)
+    {
+        SubscriptionMatchedStatus callback_status;
+        if (ReturnCode_t::RETCODE_OK == data_reader_->get_subscription_matched_status(callback_status))
+        {
+            listener->on_subscription_matched(data_reader_->user_datareader_, callback_status);
+        }
+    } else {
```

Comment:
We should please the linters:

```suggestion
    }
    else
    {
```

Reply:
Updated

Reply:
Sorry for late response.   
I updated code and please review again @MiguelCompany